### ISSUE TITLE
[Docs] Doc example of DeviceSpec doesn't work with tf 2.0

### ISSUE THREAD

**Issue by VoVAllen** [tensorflow/tensorflow#34124]
Doc link: https://www.tensorflow.org/api_docs/python/tf/DeviceSpec
```python
import tensorflow as tf
device_spec = tf.DeviceSpec(job="ps", device_type="CPU", device_index=0)
with tf.device(device_spec):
  pass
```
Got error:
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/context.py in __enter__(self)
   1508     try:
-> 1509       new_device_name, new_device_spec = _device_parsing_cache[cache_key]
   1510     except TypeError:

KeyError: ('', <tensorflow.python.framework.device_spec.DeviceSpecV2 object at 0x7f34c74de528>)

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
1 frames
/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/context.py in __enter__(self)
   1517         if not isinstance(new_device_name, six.string_types):
   1518           raise ValueError("Expecting a string device name. Got %s(%s)" %
-> 1519                            (type(new_device_name), new_device_name))
   1520         device_spec = pydev.DeviceSpec.from_string(new_device_name)
   1521         if old_device_name:

ValueError: Expecting a string device name. Got <class 'tensorflow.python.framework.device_spec.DeviceSpecV2'>(<tensorflow.python.framework.device_spec.DeviceSpecV2 object at 0x7f34c74de528>)
```

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): 2.0.0
- TensorFlow version (use command below):
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

**Describe the expected behavior**

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by nikochiko** [comment]
Syntax works fine on TensorFlow 1.15.0 (all 1.x): [gist](https://colab.research.google.com/gist/nikochiko/7cefa862d487ff9d6d31c9c2034a3cb8/34124.ipynb#scrollTo=6jMLFtVeeJpX)
Throws error in TensorFlow 2.x: [gist](https://colab.research.google.com/drive/1nkoSsW4hINcp03EPwLC0NFQhpuJURUbW#scrollTo=tB54LogAgL1D)

**Comment by nikochiko** [comment]
Update: found out the problem lies in using this command with eager execution. 
Find [gist](https://colab.research.google.com/gist/nikochiko/0de00bb23b936f1ce714bf969aae5f1b/34124_fix.ipynb) here. Must be called with `spec_name.to_string()` function in eager execution.
Will update the docs. Issue can be assigned to me. Will put a PR soon.

**Comment by nikochiko** [comment]
Thank you @VoVAllen for bringing this issue to attention.

**Comment by ymodak** [comment]
This issue is fixed with latest version TF nightly build '2.1.0-dev20191111'. Thanks!

**Comment by VoVAllen** [comment]
However I would consider it as an implementation bug instead of a documentation error. Thanks for your help!
